src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/CompilerUtilities/ImmutableHashMap.cs

```diff
@@ -52,12 +51,11 @@ internal sealed class ImmutableHashMap<TKey, TValue> : IImmutableDictionary<TKey
         /// <param name="root">The root.</param>
         /// <param name="comparer">The comparer.</param>
         /// <param name="valueComparer">The value comparer.</param>
-        private ImmutableHashMap(Bucket root, IEqualityComparer<TKey> comparer, IEqualityComparer<TValue> valueComparer)
+        private ImmutableHashMap(Bucket? root, IEqualityComparer<TKey> comparer, IEqualityComparer<TValue> valueComparer)
             : this(comparer, valueComparer)
         {
-            Debug.Assert(root != null);
```

Comment:
⚠️ I'm fairly certain this assertion was incorrect (incompatible with `WithComparers`). I couldn't find any tests for this type though.